### ISSUE TITLE
Editing compressed (or encrypted) BLOBs

### ISSUE THREAD

**Issue by mfichtner** [Sequel-Ace/Sequel-Ace#2065]
When double-clicking a BLOB, Sequel Ace opens the Pop-up Sheet and offers to view the data as Text, Image, Hex, or JSON through the bar at the bottom of the sheet. What's missing is an option to interpret the BLOB as compressed data (created using MySQL's `COMPRESS()` function).

It would be great if an option was added that lets you `UNCOMPRESS()` the data before editing it, and have it run through `COMPRESS()` again automatically when done. This should probably be a checkbox, so that the uncompressed data can still be treated as either Text, Hex, or JSON.

Support for `AES_DECRYPT()` and `AES_ENCRYPT()` in a similar way would be nice, too.

**Comment by Seemone** [comment]
I'm not sure how common it is, but I'm using gzip to compress data con the client side, to save both storage space and network traffic (the python driver I'm using (asyncmy) doesn't support protocol compression).
It would be nice if the UI could detect compressed data and show the uncompressed version on the fly

**Comment by patrickteng** [comment]
Yes, this would be a great addition. Needing to query that exact row (or rows) just to view the uncompressed data is a hassle. Or is there any other options that I can use to store the compressed data (maybe different data types) where I can still view the data on the fly?

**Comment by Seemone** [comment]
I'm seeing this similar to the UUID decoding feature
